orderer/common/channelparticipation/restapi.go

```diff
@@ -9,21 +9,28 @@ package channelparticipation
 import (
 	"encoding/json"
 	"net/http"
+	"strings"
 
+	"github.com/hyperledger/fabric/common/configtx"
 	"github.com/hyperledger/fabric/common/flogging"
 	"github.com/hyperledger/fabric/orderer/common/localconfig"
 	"github.com/pkg/errors"
 )
 
 const URLBaseV1 = "/participation/v1/"
-
-type ErrorResponse struct {
-	Error string `json:"error"`
-}
+const URLBaseV1Channels = URLBaseV1 + "channels"
 
 //go:generate counterfeiter -o mocks/channel_management.go -fake-name ChannelManagement . ChannelManagement
 
 type ChannelManagement interface {
+	// ListAll returns the names of all channels (including the system channel), and the name of the system channel
+	// (empty if does not exist). The URL field is empty, and is to be completed by the caller.
+	ListAllChannels() ([]ChannelInfoShort, string)
```

Comment:
IMO, instead of returning a touple it's better to wrap it into struct.

Reply:
Also I think that better name just to call it `Channels`

Reply:
OK